content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -147,6 +147,30 @@ const xprisma = prisma.$extends({
 })
 ```
 
+### Mutate the result of a query
+
+You can use `await` and then mutate the result of the `query` promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findFirst({ model, operation, args, query }) {
+        const user = await query(args)
+        
+        if (user.password !== undefined) {
+          user.password = '******'
+        }
+
+        return user
+      },
+    },
+  },
+})
+```
+
+While we want to show this is possible, for performance reasons, we recommend that you use the **`result` component to override existing fields**.
```

Comment:
What are these performance reasons?

Reply:
I've added this reason to the admonition